From 45e00ab18b31822bc032c57bfe76a4c759f3a789 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Fri, 27 Oct 2017 19:05:00 +0800
Subject: [PATCH 3071/5242] MLK-16715-5 usb: chipidea: imx: using phy_mode to
 judge HSIC controller

commit  30c6a67e7c8beac7653c4da150b0df986f93e76f from
https://source.codeaurora.org/external/imx/linux-imx.git

HSIC controller must use HSIC phy mode, it is more suitable way
to judge HSIC controller.

Acked-by: Jun Li <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/ci_hdrc_imx.c |   30 ++----------------------------
 1 file changed, 2 insertions(+), 28 deletions(-)

diff --git a/drivers/usb/chipidea/ci_hdrc_imx.c b/drivers/usb/chipidea/ci_hdrc_imx.c
index e6d1677..94483a1 100644
--- a/drivers/usb/chipidea/ci_hdrc_imx.c
+++ b/drivers/usb/chipidea/ci_hdrc_imx.c
@@ -20,6 +20,7 @@
 #include <linux/regulator/consumer.h>
 #include <linux/busfreq-imx.h>
 #include <linux/pm_qos.h>
+#include <linux/usb/of.h>
 
 #include "ci.h"
 #include "ci_hdrc_imx.h"
@@ -133,32 +134,6 @@ struct ci_hdrc_imx_data {
 	POWER_SUPPLY_PROP_CURRENT_MAX,	/* Maximum current in mA */
 };
 
-static inline bool is_imx6q_con(struct ci_hdrc_imx_data *imx_data)
-{
-	return imx_data->data == &imx6q_usb_data;
-}
-
-static inline bool is_imx6sl_con(struct ci_hdrc_imx_data *imx_data)
-{
-	return imx_data->data == &imx6sl_usb_data;
-}
-
-static inline bool is_imx6sx_con(struct ci_hdrc_imx_data *imx_data)
-{
-	return imx_data->data == &imx6sx_usb_data;
-}
-
-static inline bool is_imx7d_con(struct ci_hdrc_imx_data *imx_data)
-{
-	return imx_data->data == &imx7d_usb_data;
-}
-
-static inline bool imx_has_hsic_con(struct ci_hdrc_imx_data *imx_data)
-{
-	return is_imx6q_con(imx_data) ||  is_imx6sl_con(imx_data)
-		|| is_imx6sx_con(imx_data) || is_imx7d_con(imx_data);
-}
-
 /* Common functions shared by usbmisc drivers */
 
 static struct imx_usbmisc_data *usbmisc_get_init_data(struct device *dev)
@@ -561,8 +536,7 @@ static int ci_hdrc_imx_probe(struct platform_device *pdev)
 	if (pdata.flags & CI_HDRC_SUPPORTS_RUNTIME_PM)
 		data->supports_runtime_pm = true;
 
-	if (data->usbmisc_data && data->usbmisc_data->index > 1
-	    && (imx_has_hsic_con(data))) {
+	if (of_usb_get_phy_mode(dev->of_node) == USBPHY_INTERFACE_MODE_HSIC) {
 		pdata.flags |= CI_HDRC_IMX_IS_HSIC;
 		data->hsic_pad_regulator = devm_regulator_get(dev, "pad");
 		if (PTR_ERR(data->hsic_pad_regulator) == -EPROBE_DEFER) {
-- 
1.7.9.5

